apps/console/src/features/claims/api/claims.ts

```diff
@@ -647,3 +654,44 @@ export const getAllExternalClaims = (dialectID: string, params: ClaimsGetParams)
             return Promise.resolve([]);
         });
 };
+
+/**
+ * Get all SCIM resource types.
+ * 
+ * @returns response.
+ */
+export const getSCIMResourceTypes = (): Promise<any> => {
```

Comment:
Shall we use SWR here?

Reply:
I encountered challenges because we have dependent API requests, where one request relies on the data from a prior request. These requests are triggered sequentially when a button is clicked. Due to React's `setState` being asynchronous, sometimes the state doesn't update in time for the subsequent request, leading to inconsistencies.